spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -339,13 +341,14 @@ public void doManualAck(LinkedList<MessageHeaders> messageHeaders) {
 
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
-		static MimeType mimeTypeFromObject(Object payload) {
+		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.TEXT_PLAIN;
+				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
+						: MimeTypeUtils.TEXT_PLAIN;
```

Comment:
I think I'm fine with the solution, but I wonder if we should do the same for some other general String-based content types, e.g. `application/xml`.
Feel free to merge, if that.
That was just an info to share and think about.